internal/image/filemap.go

```diff
@@ -0,0 +1,46 @@
+package image
+
+import (
+	"io/fs"
+	"os"
+
+	"github.com/dghubble/trie"
+)
+
+type FileType int
+
+const (
+	RegularFile FileType = iota
+	Dir
+)
+
+type FileNode struct {
+	fileType         FileType
+	isWhiteout       bool
+	absoluteDiskPath string
+	virtualPath      string
+}
+
+type FileMap struct {
+	fileNodeTrie *trie.PathTrie
```

Comment:
Also, how much memory do we expect to save on average with using a trie structure here?

Reply:
I can check, but I doubt it's a significant amount, this is mainly used to allow us to iterate through directories.

Reply:
Can you explain a bit more what you mean by "iterate through directories"?

Reply:
E.g. if we want to scan a node_modules folder, we need to iterate through the contents of the directory. 

It's mainly used for when we want to refactor and implement the fs.FS interface.

Reply:
got it! this trie dependency is fine for now, but if memory usage differences is negligible I wonder if this can be just simplified to a sorted array of paths that we can binary search on to enable fast lookups and directory listings.

Reply:
Looks like most of the memory usage is from extracting the tar.